GG/src/DeferredLayout.cpp

```diff
@@ -0,0 +1,82 @@
+/* GG is a GUI for SDL and OpenGL.
+   Copyright (C) 2003-2008 T. Zachary Laine
+
+   This library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public License
+   as published by the Free Software Foundation; either version 2.1
+   of the License, or (at your option) any later version.
+
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with this library; if not, write to the Free
+   Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+   02111-1307 USA
+
+   If you do not wish to comply with the terms of the LGPL please
+   contact the author as other terms are available for a fee.
+
+   Zach Laine
+   whatwasthataddress@gmail.com */
+
+#include <GG/DeferredLayout.h>
+
+using namespace GG;
+
+DeferredLayout::DeferredLayout(X x, Y y, X w, Y h, std::size_t rows, std::size_t columns,
+               unsigned int border_margin/* = 0*/, unsigned int cell_margin/* = INVALID_CELL_MARGIN*/) :
+    Layout(x, y, w, h, rows, columns, border_margin, cell_margin),
+    m_ul_prerender(Pt(x, y)),
+    m_lr_prerender(Pt(x + w, y + h)),
+    m_stop_deferred_resize_recursion(false)
+{ }
+
+void DeferredLayout::SizeMove(const Pt& ul, const Pt& lr)
+{
+    if (m_stop_deferred_resize_recursion)
+        return;
+
+    RequirePreRender();
+
+    m_ul_prerender = ul;
```

Comment:
Layout is derived from Wnd, but you never seem to set the m_upperleft and m_lowerright members... Intentional? Will this break things if a layout is assumed by a parent class to have these properties be valid?